common/src/test/scala/com/nvidia/spark/rapids/ThreadFactoryBuilderTest.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.concurrent.{Callable, Executors, ExecutorService}
+
+import org.scalatest.FunSuite
+
+class ThreadFactoryBuilderTest extends FunSuite {
+
+  test("test thread factory builder") {
+    var pool1: ExecutorService = null
+    try {
+      pool1 = Executors.newFixedThreadPool(2,
+        new ThreadFactoryBuilder().setNameFormat("thread-pool1-1 %s").setDaemon(true).build())
```

Comment:
I found it odd that the code below looks exactly like this code.  We're testing setting `setDaemon` to the default value, which means if the code completely botched the handling of that setting and the method did nothing, this test would still pass.

Reply:
The default value of setDaemon is false.
I've added the cases for both setDaemon(true) and setDaemon(false)